### ISSUE TITLE
64-bit SDL 1.2 Mac builds show black screen when fullscreened

### ISSUE THREAD

**Issue by rversteegen** [ohrrpgce/ohrrpgce#1138]
Willy Elektrix [reported](https://www.slimesalad.com/forum/viewtopic.php?t=8031) that the 64-bit SDL 1.2 Mac build of Gorgonzola shows a black screen when run fullscreen. I think the problem is specific to the monitor resolution (1680*1050) and game resolution (320x200).

```
 0.3  OHRRPGCE gorgonzola 20200502.11799 gfx_sdl/music_sdl FreeBASIC 1.06.0 (04-29-2016) GCC 4.7.4 x86_64  Built on loyalist.local -gen gcc Mac OS X/Darwin 64-bit
...
 0.3  Runtime info:   music_sdl, SDL 1.2.14, SDL_Mixer 1.2.11  Mac OS 10.12.6 
...
20.5  lock_resolution()
20.5  setvideomode zoom=2 w*h = 640*400
20.5  gfx_sdl: created screensurface size=640*400 depth=32 flags=4 R=ff00 G=ff0000 B=ff000000 A=0 is_RGBcolor=0
20.5  Desktop resolution: 1680*1050
20.5  automatic_scale_factor(0.5), screen size: 1680*1050
20.5  Setting graphics scaling to x3 change_windowsize=-1
20.5  set_zoom change_windowsize=-1, zoom=3 old size = 640,400
20.5  recenter_window_hint()
20.5  setvideomode zoom=3 w*h = 960*600
20.5  gfx_sdl: created screensurface size=960*600 depth=32 flags=4 R=ff00 G=ff0000 B=ff000000 A=0 is_RGBcolor=0
20.5  /Users/AHagen/Library/Application Support/OHRRPGCE/asquatter/gameconfig.ini : gfx.fullscreen = -1, genFullscreen = 0
20.5  setvideomode zoom=3 w*h = 960*600
20.9  ! Failed to open display (windowed = 0): Failed to find display resolution: 1024x640x32
20.9  setwindowed: fallback to previous zoom
20.9  setvideomode zoom=3 w*h = 960*600
```

It looks like the Mac build machine is using a build of SDL 1.2.14 (released 2009) instead of the last release, 1.2.15 (released 2011). I notice that the error ("Failed to find display resolution") is thrown only from Mac-specific code in SDL (QZ_SetVideoFullScreen) that was rewritten in SDL 1.2.15 (http://lists.libsdl.org/pipermail/commits-libsdl.org/2011-August/004236.html). This makes me think that upgrading to SDL 1.2.15 might fix it, although 1.2.15 still has a code path to throw the same error.

(Note: x86 gfx_sdl builds use the copy of SDL in mac/Frameworks.tar.gz, while x86_64 builds use the system-installed copy of SDL.framework)

Googling the error, I found a couple mentions of it in Mac game from many years ago. I assume it's also fixed in SDL 2.

**Comment by rversteegen** [comment]
Ravancloak (who has a 1280 x 800 monitor) reports the same problem, saying 64-bit gfx_sdl builds don't work, whilc 32-bit gfx_sdl ones do. And that it happens in general for all games.

**Comment by rversteegen** [comment]
Ravancloak reports this bug is now fixed!

**Comment by rversteegen** [comment]
I note that kylekrack also mentioned that on Mac fullscreen didn't work with SDL 1.2 but does with SDL 2. So this was a very widespread problem and a good reason to switch to gfx_sdl2.

**Comment by rversteegen** [comment]
(Err... my comment from 5 days ago was made without realising that this bug is actually fixed, thanks to upgrading SDL 1.2 on the nightly build machine)
